coderd/notifications/fetcher_internal_test.go

```diff
@@ -0,0 +1,232 @@
+package notifications
+
+import (
+	"context"
+	"database/sql"
+	"testing"
+	"text/template"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
+	"golang.org/x/xerrors"
+
+	"github.com/coder/coder/v2/coderd/database/dbmock"
+)
+
+func TestNotifier_FetchHelpers(t *testing.T) {
+	t.Parallel()
+
+	t.Run("ok", func(t *testing.T) {
+		t.Parallel()
+
+		ctrl := gomock.NewController(t)
+		dbmock := dbmock.NewMockStore(ctrl)
+
+		n := &notifier{
+			store:   dbmock,
+			helpers: template.FuncMap{},
+		}
+
+		dbmock.EXPECT().GetApplicationName(gomock.Any()).Return("ACME Inc.", nil)
+		dbmock.EXPECT().GetLogoURL(gomock.Any()).Return("https://example.com/logo.png", nil)
+
+		ctx := context.Background()
+		helpers, err := n.fetchHelpers(ctx)
+		require.NoError(t, err)
+
+		appName, ok := helpers["app_name"].(func() string)
+		require.True(t, ok)
+		require.Equal(t, "ACME Inc.", appName())
+
+		logoURL, ok := helpers["logo_url"].(func() string)
+		require.True(t, ok)
+		require.Equal(t, "https://example.com/logo.png", logoURL())
+	})
+
+	t.Run("failed to fetch app name", func(t *testing.T) {
+		t.Parallel()
+
+		ctrl := gomock.NewController(t)
+		dbmock := dbmock.NewMockStore(ctrl)
+
+		n := &notifier{
+			store:   dbmock,
+			helpers: template.FuncMap{},
+		}
+
+		dbmock.EXPECT().GetApplicationName(gomock.Any()).Return("", xerrors.New("internal error"))
+
+		ctx := context.Background()
+		_, err := n.fetchHelpers(ctx)
+		require.Error(t, err)
```

Comment:
nit: I think you should be able to drop `require.Error` if you're checking `require.ErrorContains`:

- This fails if `assert.ErrorContains` returns false:  https://github.com/stretchr/testify/blob/master/require/require.go#L344-L346
- This returns false if there is no error: https://github.com/stretchr/testify/blob/master/assert/assertions.go#L1627-L1629